scripts/optIn.js

```diff
@@ -0,0 +1,48 @@
+// Do this as the first thing so that any code reading it knows the right env.
+process.env.BABEL_ENV = 'test';
+process.env.NODE_ENV = 'test';
+process.env.PUBLIC_URL = '';
+
+const readline = require('readline');
+const { execSync } = require('child_process');
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+new Promise(function(resolve, reject) {
+  rl.question('[PRECOMMIT] Run all test suites? (Y/n) ', function(answer) {
+    if (answer.toLowerCase() === 'y') {
+      execSync('npm run test -- precommit', { stdio: 'inherit' }, function(
+        error,
+        stdout,
+        stderr
+      ) {
+        if (error) {
+          console.log('ERR: Exiting with code', error.code);
+          console.log(stderr.toString());
+        }
+      });
+    }
+    rl.question('[PRECOMMIT] Test optimized production build? (Y/n) ', function(
```

Comment:
Seems like there is a decent amount of repeated code here. What about making a function `executeCommand` where you pass in the prompt and script?